src/main/java/com/coveros/selenified/utilities/Property.java

```diff
@@ -153,13 +158,39 @@ public static boolean isProxySet() {
      *
      * @return String: the set proxy address, null if none are set
      */
-    public static String getProxy() {
+    public static String getProxy() throws InvalidProxyException {
         if (!isProxySet()) {
             return null;
         }
+        String proxy = getProgramProperty(PROXY);
+        String[] proxyParts = proxy.split(":");
+        if (proxyParts.length != 2) {
+            throw new InvalidProxyException("Proxy '" + proxy + "' isn't valid. Must contain address and port, without protocol");
+        }
+        try {
+            Integer.parseInt(proxyParts[1]);
+        } catch (NumberFormatException e) {
+            throw new InvalidProxyException("Proxy '" + proxy + "' isn't valid. Must contain address and port, without protocol. Invalid port provided. " + e);
+        }
         return getProgramProperty(PROXY);
     }
 
+    public static String getProxyHost() throws InvalidProxyException {
+        if (!isProxySet()) {
+            return null;
+        }
+        String proxy = getProxy();
+        return proxy.split(":")[0];
+    }
+
+    public static int getProxyPort() throws InvalidProxyException {
+        if (!isProxySet()) {
+            return 0;
```

Comment:
why not throw InvalidProxyException here? And in the previous method instead of returning null.